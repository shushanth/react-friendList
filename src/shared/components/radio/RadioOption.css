:local(.radioOption) {
  align-items: center;
  cursor: pointer;
  display: inherit;
  font-family: inherit;
  font-size: inherit;
}

:local(.radioOption input) {
  margin: 0;
}

:local(.radioOption label) {
  color: #fff;
  cursor: pointer;
  font-weight: 400;
  line-height: 2;
  letter-spacing: 0.1em;
  margin: 0 0.5em -0.1em;
  padding: 0;
}
