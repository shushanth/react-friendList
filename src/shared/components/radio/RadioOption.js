/**
 * Generic component for radio option, used in composed component (Radio group)
 */

import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

import styles from "./RadioOption.css";

const RadioOption = ({ name, isChecked, value, onChange }) => {
  return (
    <div className={classnames(styles.radioOption)}>
      <input
        className={classnames(styles.radioOption)}
        type="radio"
        value={value}
        id={value}
        name={name}
        checked={isChecked}
        onChange={onChange}
      />
      <label htmlFor={value}>{value}</label>
    </div>
  );
};

RadioOption.PropTypes = {
  name: PropTypes.string,
  checked: PropTypes.bool,
  onChange: PropTypes.func
};

export default RadioOption;
