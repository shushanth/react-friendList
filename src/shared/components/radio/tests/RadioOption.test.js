import React from "react";
import { shallow } from "enzyme";

import RadioOption from "../RadioOption";

describe("Component: <RadioOption/>", () => {
  let props;
  let RadioOptionWrapper;
  beforeAll(() => {
    props = {
      name: "test",
      isChecked: false,
      value: "Test",
      onChange: jest.fn()
    };
    RadioOptionWrapper = shallow(<RadioOption {...props} />);
  });

  it("should render without crashing", () => {
    expect(RadioOptionWrapper.exists()).toBe(true);
    expect(RadioOptionWrapper).toMatchSnapshot();
  });

  it("should have input with attributes , matches the supplied props values", () => {
    expect(RadioOptionWrapper.find("input").props().value).toEqual(props.value);
    expect(RadioOptionWrapper.find("input").props().checked).toEqual(props.isChecked);
    expect(RadioOptionWrapper.find("input").props().name).toEqual(props.name);
  });

  it("should have label value same as prop value", () => {
    expect(RadioOptionWrapper.find("label").props().htmlFor).toEqual(props.value);
    expect(RadioOptionWrapper.find("label").text()).toEqual(props.value);
  });
  
  describe("method: onChange", () => {
    it("should invoke onChange supplied prop method, on change in input", () => {
      let inputRadioOption = RadioOptionWrapper.find("input");
      inputRadioOption.simulate('change');
      expect(props.onChange).toHaveBeenCalled();
    })
  })

});
