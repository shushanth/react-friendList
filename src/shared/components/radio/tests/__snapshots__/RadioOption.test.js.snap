exports[`Component: <RadioOption/> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioOption
    isChecked={false}
    name="test"
    onChange={[Function]}
    value="Test" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          checked={false}
          className=""
          id="Test"
          name="test"
          onChange={[Function]}
          type="radio"
          value="Test" />,
        <label
          htmlFor="Test">
          Test
</label>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "className": "",
          "id": "Test",
          "name": "test",
          "onChange": [Function],
          "type": "radio",
          "value": "Test",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
          "htmlFor": "Test",
        },
        "ref": null,
        "rendered": "Test",
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={false}
            className=""
            id="Test"
            name="test"
            onChange={[Function]}
            type="radio"
            value="Test" />,
          <label
            htmlFor="Test">
            Test
</label>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "className": "",
            "id": "Test",
            "name": "test",
            "onChange": [Function],
            "type": "radio",
            "value": "Test",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
            "htmlFor": "Test",
          },
          "ref": null,
          "rendered": "Test",
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
